Menus/UndoMenu/UndoMenu.cs

```diff
@@ -0,0 +1,204 @@
+        const string k_FeedbackHintForJoystickController = "Click + flick left/right to undo/redo";
+        const string k_FeedbackHintForTrackpadController = "Click left/right side to undo/redo";
+
+        [SerializeField]
+        ActionMap m_ActionMap;
+
+        [SerializeField]
+        UndoMenuUI m_UndoMenuPrefab;
+
+        [SerializeField]
+        HapticPulse m_UndoPulse;
+
+        UndoMenuUI m_UndoMenuUI;
+        List<ActionMenuData> m_MenuActions;
+        Transform m_AlternateMenuOrigin;
+        MenuHideFlags m_MenuHideFlags = MenuHideFlags.Hidden;
+        float m_PrevNavigateX;
+        bool m_StillEngagedAfterStickRelease;
+        Coroutine m_StillEngagedAfterStickReleasedCoroutine;
+        bool m_TrackpadController;
+        string m_FeedbackHintForCurrentController;
+
+        readonly BindingDictionary m_Controls = new BindingDictionary();
+        
+        public Transform rayOrigin { private get; set; }
+
+        public Transform menuOrigin { get; set; }
+
+        public GameObject menuContent { get { return m_UndoMenuUI.gameObject; } }
+
+        public Node node { get; set; }
+
+        public event Action<Transform> itemWasSelected;
+
+        public Bounds localBounds { get { return default(Bounds); } }
+
+        public ActionMap actionMap { get { return m_ActionMap; } }
+        public bool ignoreLocking { get { return false; } }
+
+        public List<ActionMenuData> menuActions
+        {
+            get { return m_MenuActions; }
+            set
+            {
+                m_MenuActions = value;
+
+                if (m_UndoMenuUI)
+                    m_UndoMenuUI.actions = value;
+            }
+        }
+
+        public Transform alternateMenuOrigin
+        {
+            get { return m_AlternateMenuOrigin; }
+            set
+            {
+                m_AlternateMenuOrigin = value;
+
+                if (m_UndoMenuUI != null)
+                    m_UndoMenuUI.alternateMenuOrigin = value;
+            }
+        }
+
+        public MenuHideFlags menuHideFlags
+        {
+            get { return m_MenuHideFlags; }
+            set
+            {
+                if (m_MenuHideFlags != value)
+                {
+                    m_MenuHideFlags = value;
+
+                    var visible = value == 0;
+                    if (m_UndoMenuUI)
+                        m_UndoMenuUI.visible = visible;
+
+                    if (visible)
+                        ShowFeedback();
+                    else
+                        this.ClearFeedbackRequests();
+                }
+            }
+        }
+
+        void Start()
+        {
+            m_UndoMenuUI = this.InstantiateUI(m_UndoMenuPrefab.gameObject).GetComponent<UndoMenuUI>();
+            m_UndoMenuUI.alternateMenuOrigin = alternateMenuOrigin;
+            m_UndoMenuUI.actions = menuActions;
+            this.ConnectInterfaces(m_UndoMenuUI); // Connect interfaces before performing setup on the UI
+            m_UndoMenuUI.Setup();
```

Comment:
Remove setup method, just disable UI gameobject here.

Reply:
Yep! Though we use them in quite a few places, our goal is to avoid Setup methods where possible.